### ISSUE TITLE
software low res demo crash

### ISSUE THREAD

**Issue by pa-august** [nukeykt/NBlood#401]
- Crash to desktop occurs as Caleb hops out of his coffin in the opening demo.

- Crash only occurs in software mode at resolutions of 800x600 and below. Low res hardware modes do not crash.

- Beginning a single player mission before the demo gets to that point prevents the crash. You can start Cradle to Grave and hop out Caleb's coffin yourself without indecent.

My hardware is an Intel NUC with the latest Intel video drivers. Monitor is 1080p.

**Comment by Hendricks266** [comment]
Thanks for the report. Confirmed. This is the backtrace I get:

```
(gdb) bt
#0  0x00000000007be9c5 in MV_MixStereo<unsigned char, short> (voice=0x24df8520, length=240)
    at source/build/include/fix16.h:98
#1  0x000000000059aedf in MV_Mix (voice=voice@entry=0x24df8520, buffer=<optimized out>)
    at source/audiolib/src/multivoc.cpp:149
#2  0x000000000059b5af in MV_ServiceVoc () at source/audiolib/src/multivoc.cpp:286
#3  0x00000000005a2d67 in fillData (userdata=<optimized out>,
    ptr=0x14e8eff0
    at source/audiolib/src/driver_sdl.cpp:80
#4  0x00000000006bb92e in SDL_RunAudio (devicep=devicep@entry=0x157240) at ../src/audio/SDL_audio.c:735
#5  0x0000000000679284 in SDL_RunThread (thread=0x156e98) at ../src/thread/SDL_thread.c:276
#6  0x00000000006b7c51 in RunThread (data=<optimized out>) at ../src/thread/windows/SDL_systhread.c:82
#7  RunThreadViaCreateThread (data=<optimized out>) at ../src/thread/windows/SDL_systhread.c:92
#8  0x00007ffae6896fd4 in KERNEL32!BaseThreadInitThunk () from C:\WINDOWS\System32\kernel32.dll
#9  0x00007ffae795cec1 in ntdll!RtlUserThreadStart () from C:\WINDOWS\SYSTEM32\ntdll.dll
#10 0x0000000000000000 in ?? ()
Backtrace stopped: previous frame inner to this frame (corrupt stack?)
(gdb) info locals
isample0 = <optimized out>
source = 0x1b9 <error: Cannot access memory at address 0x1b9>
dest = 0x24dfb780
position = 4269165568
rate = 0
volume = <optimized out>
(gdb) f 1
#1  0x000000000059aedf in MV_Mix (voice=voice@entry=0x24df8520, buffer=<optimized out>)
    at source/audiolib/src/multivoc.cpp:149
149             voice->position = voice->mix(voice, mixlen);
(gdb) info locals
mixlen = 256
position = <optimized out>
voclen = 1060700160
gv = 65535
length = 256
bufsiz = 4078197792
rate = 2087683040
(gdb) f 2
#2  0x000000000059b5af in MV_ServiceVoc () at source/audiolib/src/multivoc.cpp:286
286                 if (!MV_Mix(voice, MV_MixPage))
(gdb) info locals
voice = 0x24df8520
next = 0x24df8478
MusicVoice = 0x0
```

**Comment by CommonLoon102** [comment]
I can reproduce it too on Windows 7, while debugging in Visual Studio.

**Comment by nukeykt** [comment]
Should be fixed in 41c8b603947ce48029da64d7600d2fcc9d5aa7c1
